modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheContextInfo.java

```diff
@@ -53,7 +55,25 @@
      * @param clientCache Client cache or not.
      */
     public GridCacheContextInfo(GridCacheContext<K, V> cctx, boolean clientCache) {
-        config = cctx.config();
+        config = new CacheConfiguration(cctx.config());
+        dynamicDeploymentId = null;
+        groupId = cctx.groupId();
+        cacheId = cctx.cacheId();
+
+        this.clientCache = clientCache;
+
+        this.cctx = cctx;
+    }
+
+    /**
+     * Constructor of full cache context.
+     *
+     * @param cctx Cache context.
+     * @param config Cache configuration, that may be different from cache context.
+     * @param clientCache Client cache or not.
+     */
+    public GridCacheContextInfo(GridCacheContext<K, V> cctx, CacheConfiguration<K, V> config, boolean clientCache) {
+        this.config = new CacheConfiguration(config);
```

Comment:
Let's use `new CacheConfiguration<>` to remove compiler warning

Reply:
See comment above.